goplay/src/main/java/be/tapped/goplay/epg/EpgRepo.kt

```diff
@@ -4,74 +4,57 @@ import arrow.core.Either
 import arrow.core.computations.either
 import arrow.core.left
 import arrow.core.right
-import be.tapped.common.internal.executeAsync
 import be.tapped.goplay.ApiResponse
-import be.tapped.goplay.common.goPlayApiDefaultOkHttpClient
-import be.tapped.goplay.common.safeBodyString
+import be.tapped.goplay.common.ktorClient
 import be.tapped.goplay.common.siteUrl
+import io.ktor.client.HttpClient
+import io.ktor.client.request.get
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
-import kotlinx.serialization.decodeFromString
-import kotlinx.serialization.json.Json
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import java.text.SimpleDateFormat
-import java.util.Calendar
-import java.util.TimeZone
 
-public class JsonEpgParser {
-    public fun parse(json: String): Either<ApiResponse.Failure, List<EpgProgram>> = Either.catch {
-        Json {
-            isLenient = true
-            ignoreUnknownKeys = true
-        }.decodeFromString<List<EpgProgram>>(json)
-    }.mapLeft(ApiResponse.Failure::JsonParsingException)
-}
-
-public sealed interface EpgRepo {
+public fun interface EpgRepo {
 
     public enum class Brand {
         VIER,
         VIJF,
         ZES;
     }
 
+    /**
+     * @param month 0 based. 0 being january
+     * @param dayOfTheMonth 1-31 (Arrow Analysis?)
+     */
+    public data class Date(val year: Int, val month: Int, val dayOfTheMonth: Int)
+
     public suspend fun epg(
         brand: Brand,
-        calendar: Calendar = Calendar.getInstance(TimeZone.getTimeZone("Europe/Brussels"))
+        date: Date,
     ): Either<ApiResponse.Failure, ApiResponse.Success.ProgramGuide>
 }
 
-public class HttpEpgRepo(
-    private val client: OkHttpClient = goPlayApiDefaultOkHttpClient,
-    private val jsonEpgParser: JsonEpgParser = JsonEpgParser(),
-    private val dateFormatter: SimpleDateFormat = SimpleDateFormat("yyyy-MM-dd"),
-) : EpgRepo {
+// curl -X GET "https://www.goplay.be/api/epg/vier/2020-12-13"
+// curl -X GET "https://www.goplay.be/api/epg/vijf/2020-12-13"
+// curl -X GET "https://www.goplay.be/api/epg/zes/2020-12-13"
+internal fun defaultEpgRepo(client: HttpClient = ktorClient): EpgRepo =
+    EpgRepo { brand: EpgRepo.Brand, date: EpgRepo.Date ->
+        val (year, month, day) = date
+        val brandPath = when (brand) {
+            EpgRepo.Brand.VIER -> "vier"
+            EpgRepo.Brand.VIJF -> "vijf"
+            EpgRepo.Brand.ZES -> "zes"
+        }
+        val monthStr = if (month < 10) "0${month}" else "$month"
```

Comment:
🤮
Is there a Kotlin only time lib?

Reply:
https://github.com/Kotlin/kotlinx-datetime

Reply:
It's on their current roadmap to become completely stable soon.

Reply:
I also looked further into this lib, but it seems for now it does not have a formatter option which is exactly what we were looking for.